pilot/pkg/serviceregistry/kube/controller/pod.go

```diff
@@ -149,7 +149,14 @@ func (pc *PodCache) onEvent(old, pod *v1.Pod, ev model.Event) error {
 	// PodIP will be empty when pod is just created, but before the IP is assigned
 	// via UpdateStatus.
 	if len(ip) == 0 {
-		return nil
+		// However, in the case of an Eviction, the event that marks the pod as Failed may *also* have removed the IP.
+		// If the pod *used to* have an IP, then we need to actually delete it.
+		if ip = pc.getIPByPod(config.NamespacedName(pod)); len(ip) > 0 {
```

Comment:
Would it be easier to read like
```
ip = pc.getIPByPod(...)
if len(ip) > 0 { ... }
```

Given that `ip` is being reassigned here?

Reply:
(seems like there's already a precedent for overwriting variables in the codebase), and given that IP is already empty/null, IMO this is fine?

Reply:
SGTM